packages/java/plugin-base/src/main/java/dev/hilla/plugin/base/HillaAppInitUtility.java

```diff
@@ -159,8 +159,13 @@ private static String findSpringBootApplicationPackage(
     private static String extractPackage(Path path) {
         try {
             var content = Files.readString(path);
-            var matcher = Pattern.compile("^\\s*package\\s+([\\w.]+)\\s*;")
-                    .matcher(content);
+            var regexToExcludeCommentsAndStringLiterals = "//.*|(\"(?:\\\\[^\"]|\\\\\"|.)*?\")|(?s)/\\*.*?\\*/";
```

Comment:
I was thinking that, instead of taking risks with regexps, we can also get the package from the relative path.

Reply:
Working with paths also can work, however, with paths we need to consider skipping the `/src/main/java/` part or `src\main\java` in case of Windows. Then we might consider keeping it as Kotlin friendly as possible by considering to skip `src/main/kotlin` or `src\main\kotlin` for Windows, and I'd say it is not much more convenient either. (Supporting Kotlin is not our priority at the moment, anyway).

I'm not a fan of Regex myself, however, I'd prefer it over dealing with OS dependent paths :) 
I think there is not much of a risk if we have enough tests in place to make sure it works all known corner cases.